src/snowflake/snowpark/mock/_plan.py

```diff
@@ -443,22 +468,32 @@ def handle_udf_expression(
     res.name = quote_name(f"{exp.udf_name}({', '.join(input_data.columns)})".upper())
 
     if udf_name in analyzer.session.udf._udf_level_imports:
+        # Remove UDF level imports
         for module_path in analyzer.session.udf._udf_level_imports[udf_name]:
             if (
                 module_path in sys.path
-            ):  # We need this additional check since children expression might temper with sys.path as well, this needs some future work
+            ):  # TODO: For now we need this additional check since children expression can write to sys.path as well
```

Comment:
When will this no longer be the case?

Reply:
When we work on adding more isolation, e.g. run UDF in a separate process.